### PR TITLE
Api create patient

### PR DESCRIPTION
**Pull request by basicem**

### CODE REVIEW

File: controllers/patient.controller.js [hunk 1]
```diff
@@ -15,9 +16,13 @@ const get = async (req, res) => {
 
 const post = async (req, res) => {
   try {
-    const patient = await patientServices.createPatient(req.body);
+    const value = await patientSchema.validateAsync(req.body);
+    const patient = await patientServices.createPatient(value);
     return res.status(201).json({ id: patient.id });
   } catch (err) {
+    if (err.isJoi) {
+      return res.status(401).json({ error: err.details });
```
Comment (amarbasic) [inline controllers/patient.controller.js]: Why `401`? 401 status code is when user is not authenticated.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

File: schemas/patient.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const Joi = require("joi");
+
+const patientSchema = Joi.object({
+  image: Joi.string().required(),
+  firstName: Joi.string().required(),
```
Comment (amarbasic) [inline schemas/patient.js:5]: Let's add `dateOfBirth`

**Review by amarbasic — COMMENTED** [review]
We need tests for post patient endpoint

File: schemas/patient.js (hunk 2 above)
Comment (basicem) [reply schemas/patient.js:5]: ok

File: controllers/patient.controller.js [hunk 3]
```diff
@@ -15,9 +16,13 @@ const get = async (req, res) => {
 
 const post = async (req, res) => {
   try {
-    const patient = await patientServices.createPatient(req.body);
+    const value = await patientSchema.validateAsync(req.body);
+    const patient = await patientServices.createPatient(value);
     return res.status(201).json({ id: patient.id });
   } catch (err) {
+    if (err.isJoi) {
```
Comment (basicem) [inline controllers/patient.controller.js]: move to error handling to helpers `resolve_error` in `helpers/controllers.js`

File: startup.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const cors = require("cors");
 
 app.use(cors());
 
-app.use(bodyParser.json());
+app.use(bodyParser.json({ limit: "50mb" }));
```
Comment (basicem) [inline startup.js]: set 5mb

File: tests/controllers/patients.test.js [hunk 5]
```diff
@@ -47,3 +49,88 @@ describe("GET patients", async () => {
...
+      dateOfBirth: "1990-01-07",
+      address: "Sarajevska 77",
+      city: "Sarajevo",
+      phoneNumber: "888-120-123",
+      email: "joedoe@gmail.com"
+    };
+    // act
+    const response1 = await request(app).post("/api/patients").send(patient1);
+    const response2 = await request(app).post("/api/patients").send(patient1);
+    const response3 = await request(app).get("/api/patients");
```
Comment (basicem) [inline tests/controllers/patients.test.js]: Test only `post` - assert using DB

File: tests/controllers/patients.test.js [hunk 6]
```diff
@@ -47,3 +49,88 @@ describe("GET patients", async () => {
...
+      lastName: "Doe",
+      dateOfBirth: "1990-01-07",
+      address: "Sarajevska 77",
+      city: "Sarajevo",
+      phoneNumber: "888-120-123",
+      email: "joedoe@gmail.com"
+    };
+    // act
+    const response1 = await request(app).post("/api/patients").send(patient1);
+    const response2 = await request(app).get("/api/patients");
```
Comment (amarbasic) [inline tests/controllers/patients.test.js]: Dont test get method in tests for post

File: tests/controllers/patients.test.js (hunk 5 above)
Comment (amarbasic) [reply tests/controllers/patients.test.js]: Dont call get here

File: tests/controllers/patients.test.js [hunk 7]
```diff
@@ -47,3 +49,88 @@ describe("GET patients", async () => {
...
+      dateOfBirth: "1990-01-07",
+      address: "Sarajevska 77",
+      city: "Sarajevo",
+      phoneNumber: "129-333-123",
+      email: "janedoe@gmail.com"
+    };
+    // act
+    const response1 = await request(app).post("/api/patients").send(patient1);
+    const response2 = await request(app).post("/api/patients").send(patient2);
+    const response3 = await request(app).get("/api/patients");
```
Comment (amarbasic) [inline tests/controllers/patients.test.js]: You should not call get here because you are testing post

**Review by amarbasic — APPROVED** [review]

**Pull request merged**